pkg/oc/cli/admin/release/git.go

```diff
@@ -29,7 +29,7 @@ var noSuchRepo = errors.New("location is not a git repo")
 func (g *git) exec(command ...string) (string, error) {
 	buf := &bytes.Buffer{}
 	bufErr := &bytes.Buffer{}
-	cmd := exec.Command("git", append([]string{"-C", g.path}, command...)...)
+	cmd := exec.Command("git", append([]string{"--git-dir", filepath.Join(g.path, ".git")}, command...)...)
```

Comment:
This works, but a potentially more reliable alternative (that also supports bare repositories) would be to set [`Dir`][1] in the constructed command.

```go
cmd := exec.Command("git", command...)
cmd.Dir = g.path
glog.V(5).Infof("Executing git in %s: %v\n", cmd.Dir, cmd.Args)
```

[1]: https://golang.org/pkg/os/exec/#Cmd

Reply:
Good point. Won't work for clone but will work for the others.